File: Samples/java/src/main/java/com/phunware/java/sample/PersonMarker.java
package com.phunware.java.sample;

import com.google.android.gms.maps.model.Marker;

/**
 * A wrapper around a maker that associates a
 */
public class PersonMarker {
    private Marker marker;
    private String name;
    private String userType;

    public PersonMarker(String name, String userType, Marker marker) {
        super();
        this.name = name;
        this.userType = userType;
        this.marker = marker;
    }

    public Marker getMarker() {
        return marker;
    }

    public String getName() {
        return name;
    }

    public String getUserType() {
        return userType;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        PersonMarker that = (PersonMarker) o;

        if (marker != null ? !marker.equals(that.marker) : that.marker != null) return false;
        if (name != null ? !name.equals(that.name) : that.name != null) return false;
        return userType != null ? userType.equals(that.userType) : that.userType == null;

    }

    @Override
    public int hashCode() {
        int result = marker != null ? marker.hashCode() : 0;
        result = 31 * result + (name != null ? name.hashCode() : 0);
        result = 31 * result + (userType != null ? userType.hashCode() : 0);
        return result;
    }
}

File: Samples/kotlin/src/main/java/com/phunware/kotlin/sample/PersonMarker.kt
package com.phunware.kotlin.sample

import com.google.android.gms.maps.model.Marker

/**
 * A wrapper around a maker that associates a
 */
class PersonMarker(val name: String?, val userType: String?, val marker: Marker?) {

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val that = o as PersonMarker?

        if (if (marker != null) marker != that!!.marker else that!!.marker != null) return false
        if (if (name != null) name != that.name else that.name != null) return false
        return if (userType != null) userType == that.userType else that.userType == null

    }

    override fun hashCode(): Int {
        var result = marker?.hashCode() ?: 0
        result = 31 * result + (name?.hashCode() ?: 0)
        result = 31 * result + (userType?.hashCode() ?: 0)
        return result
    }
}

